### DISCUSSION TITLE
Custom object mapper doesn't work

### DISCUSSION THREAD

**Discussion by QuantumBear** [Netflix/dgs-framework#1627] [Q&A]
`@Configuration
public class DgsConfiguration {

    @Bean
    @Qualifier("dgsObjectMapper")
    public ObjectMapper dgsObjectMapper() {
        ObjectMapper customMapper = new ObjectMapper();
        customMapper.setPropertyNamingStrategy(PropertyNamingStrategies.SNAKE_CASE);
        return customMapper;
    }

}`

We want to convert camel case json fields to snake case in response, but it seems not working both in graphiql interface or graphql rest api. Could anyone help me with this, thanks!

**Comment by congotej** [comment]
Hi @QuantumBear ! Looks like you are setting up the [custom object mapper](https://netflix.github.io/dgs/advanced/custom-object-mapper/) correctly. Can you try adding a unit test to validate if `PropertyNamingStrategies.SNAKE_CASE` is working as expected? This way you can confirm where the issue is.

**Comment by cesar05** [comment]
Hello @QuantumBear I have the same problem.
Could you solve it?

**Reply by srinivasankavitha** [reply]
Hi - For custom serialization, the custom object mapper needs to update Spring's object mapper as described here: https://netflix.github.io/dgs/advanced/custom-object-mapper/

More specifically for your issue of using snake case, it won't work unless you register explicit data fetchers to handle snake case to camel case conversions. More context and examples described here: https://netflix.github.io/dgs/advanced/custom-object-mapper/

**Comment by cesar05** [comment]
In the next thread is the solve it.
https://github.com/Netflix/dgs-framework/discussions/1742
